californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/ObserveServerSideTest.java

```diff
@@ -437,6 +451,40 @@ public void testQuickChangeAndTimeout() throws Exception {
 		Thread.sleep(ACK_TIMEOUT + 100);
 		assertThat("Resource has not removed observe relation", testObsResource.getObserverCount(), is(0));
 	}
+	
+	/**
+	 * Test incomplete block2 notification (missing request)
+	 * 
+	 * @throws Exception
+	 */
+	@Test
+	public void testIncompleteBlock2Notification() throws Exception {
+		System.out.println("Observe with blockwise");
+		respPayload = generateRandomPayload(32);
+		byte[] tok = generateNextToken();
+
+		// Establish observe relation
+		respType = null; // first type is normal ACK
+		client.sendRequest(CON, GET, tok, ++mid).path(RESOURCE_PATH).observe(0).go();
+		client.expectResponse(ACK, CONTENT, tok, mid).storeObserve("A").storeETag("tag").payload(respPayload).go();
+		Assert.assertEquals("Resource has not added relation:", 1, testObsResource.getObserverCount());
```

Comment:
I'm not sure, if we want to use a more uniform way, but if, I would have expected 

`assertThat("Resource has not added relation:", testObsResource.getObserverCount(), is(1)); `

Reply:
All the tests in this class is written using :
Assert.assertEquals("Resource has not added relation:", 1, testObsResource.getObserverCount());

Reply:
OK.

(Except of 2, which use the assertThat. Rise the question, which one should be used for future tests. But that's sure out of scope this PR.)